src/Effects/Chorus.cpp

```diff
@@ -131,41 +176,69 @@ void Chorus::out(const Stereo<float *> &input)
-                     + cinterpolate(delaySample.l, maxdelay,
-                                    dlhi) * (1.0f - dllo);
-        delaySample.l[dlk] = inL + efxoutl[i] * fb;
+        // linear interpolate from old to new value over length og the buffer
+        float dl = (dlHist * (buffersize - i) + dlNew * i) / buffersize_f;
+        // get sample with that delay form delay line and add to output accumulator
+        output += getSample(delaySample.l, dl, dlk);
+        switch (Pflangemode) {
+            case DUAL:
+            // calculate and apply delay for second ensemble member
+            dl = (dlHist2 * (buffersize - i) + dlNew2 * i) / buffersize_f;
+            output += getSample(delaySample.l, dl, dlk);
+                break;
+            case TRIPLE:
+            // calculate and apply delay for second ensemble member
+            dl = (dlHist2 * (buffersize - i) + dlNew2 * i) / buffersize_f;
+            output += getSample(delaySample.l, dl, dlk);
+            // same for third ensemble member
+            dl = (dlHist3 * (buffersize - i) + dlNew3 * i) / buffersize_f;
+            output += getSample(delaySample.l, dl, dlk);
+            // reduce amplitude to match single phase modes
+            output *= 0.85f;
+                break;
+            default:
+                // nothing to do for standard chorus
+                break;
+        }
+        // store current input + feedback to delay line at writing position
+        delaySample.l[dlk] = inL + output * fbComp;
+        // write output to output interface
+        efxoutl[i] = output;
 
         //Right channel
-
-        //compute the delay in samples using linear interpolation between the lfo delays
-        mdel = (dr1 * (buffersize - i) + dr2 * i) / buffersize_f;
+        output = 0.0f;
         if(++drk >= maxdelay)
             drk = 0;
-        tmp = drk * 1.0f - mdel + maxdelay * 2.0f; //where should I get the sample from
-
-        dlhi  = (int) tmp;
-        dlhi %= maxdelay;
+        float dr = (drHist * (buffersize - i) + drNew * i) / buffersize_f;
+        output += getSample(delaySample.r, dr, drk);
+        switch (Pflangemode) {
+            case DUAL:
+                // calculate and apply delay for second ensemble member
+                dr = (drHist2 * (buffersize - i) + drNew2 * i) / buffersize_f;
+                output += getSample(delaySample.r, dr, drk);
+                break;
+            case TRIPLE:
+                // calculate and apply delay for second ensemble member
+                dr = (drHist2 * (buffersize - i) + drNew2 * i) / buffersize_f;
+                output += getSample(delaySample.r, dr, drk);
+                // same for third ensemble member
+                dr = (drHist3 * (buffersize - i) + drNew3 * i) / buffersize_f;
+                output += getSample(delaySample.r, dr, drk);
+                // reduce amplitude to match single phase modes
+                output *= 0.85f;
```

Comment:
Could we, instead, add this factor to `fbComp`? (above, line 167)

Reply:
good idea